### PR TITLE
chore: release v2.3.0-mainnet

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description
This PR implements the changes needed in order to tag `v2.1.0-mainnet` of Desmos. Refer to #657 for major details.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by bragaz — APPROVED** [review]
Forgot to approve this.

**Comment by bragaz** [comment]
@RiccardoM I've solved the conflicts. Please take a look whenever you can. It should be the correct configuration since this is the first on-chain auto-upgrade of the mainnet/.

File: app/app.go [hunk 1]
```diff
@@ -689,17 +691,41 @@ func (app *DesmosApp) RegisterTendermintService(clientCtx client.Context) {
 }
 
 func (app *DesmosApp) registerUpgradeHandlers() {
-	app.upgradeKeeper.SetUpgradeHandler("v2.3.0", func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
-		return app.mm.RunMigrations(ctx, app.configurator, vm)
+	app.upgradeKeeper.SetUpgradeHandler("v2.1.0", func(ctx sdk.Context, plan upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
```
Comment (RiccardoM) [inline app/app.go:694]: The upgrade name should be `v2.3.0`

File: app/app.go [hunk 2]
```diff
@@ -689,17 +691,41 @@ func (app *DesmosApp) RegisterTendermintService(clientCtx client.Context) {
...
 	})
 
 	upgradeInfo, err := app.upgradeKeeper.ReadUpgradeInfoFromDisk()
 	if err != nil {
 		panic(err)
 	}
 
-	if upgradeInfo.Name == "v2.3.0" && !app.upgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
-		storeUpgrades := storetypes.StoreUpgrades{}
+	if upgradeInfo.Name == "v2.1.0" && !app.upgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
```
Comment (RiccardoM) [inline app/app.go:725]: Ditto

**Comment by bragaz** [comment]
@RiccardoM done

File: .github/workflows/on-chain-upgrade.yml [hunk 3]
```diff
@@ -22,9 +22,9 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 30
     env:
-      GENESIS_DESMOS_VERSION: "v2.2.1-testnet"
-      GENESIS_URL: "https://raw.githubusercontent.com/RiccardoM/desmos-states/master/morpheus-apollo-2-2721215.json"
-      UPGRADE_NAME: "v2.3.0"
+      GENESIS_DESMOS_VERSION: "v1.0.1"
+      GENESIS_URL: "https://raw.githubusercontent.com/RiccardoM/desmos-states/master/desmos-mainnet-780000.json"
+      UPGRADE_NAME: "v2.1.0"
```
Comment (RiccardoM) [inline .github/workflows/on-chain-upgrade.yml:27]:
```suggestion
      UPGRADE_NAME: "v2.3.0"
```

**Review by bragaz — APPROVED** [review]

**Comment by bragaz** [comment]
@RiccardoM looking at [this](https://docs.cosmos.network/master/building-modules/upgrade.html#consensus-version) I've seen that we need to hardcode the `ConseusVersion` to 1 the first time.
Currently it's set to 3. We should change it, confirm?

**Comment by RiccardoM** [comment]
> @RiccardoM looking at [this](https://docs.cosmos.network/master/building-modules/upgrade.html#consensus-version) I've seen that we need to hardcode the `ConseusVersion` to 1 the first time. Currently it's set to 3. We should change it, confirm?

That is not correct. The `ConsensusVersion` identifies the latest version of a module. In our case, the `x/profiles` module should be at its third version (`ConsensusVersion = 3`) due to 2 previous migrations we had to run. 

When the upgrade height comes, what happens is the following: 
- the current veision of each module is read using the `fromVm` map 
- the updated version of each module is read using the `ConsensusVersion` method
- for each module, we perform an upgrade `old_version -> current_version`

If the current version is `1` and the new one is `3`, then two upgrades are performed: 
1. From `1` to `2`
2. From `2` to `3`

And this is why inside the `x/profiles` migrator we have two methods: [`Migrate1To2`](https://github.com/desmos-labs/desmos/blob/master/x/profiles/keeper/migrations.go#L29) and [`Migrate2To3`](https://github.com/desmos-labs/desmos/blob/master/x/profiles/keeper/migrations.go#L34)

**Review by bragaz — APPROVED** [review]

**Pull request merged**